### PR TITLE
feat: use URL shortener when Share button is used

### PR DESCRIPTION
**Pull request by brandonroberts**
## Description

- Use URL shortener in OpenSauced API to generate a short URL whenever the `Share` button is used before copying to the clipboard

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes https://github.com/open-sauced/app/issues/522

## Steps to QA

1. Go to a page with a Share button (contributor insight, repository insight, contributor profile)
2. Click the Share button
3. Upon success the short URL is copied to the clipboard instead of the full page URL
4. If the API request fails for any reason, the original URL is returned

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: lib/utils/shorten-url.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export async function shortenUrl(url: string) {
+  try {
+    const response = await fetch(`${process.env.NEXT_PUBLIC_API_URL}/url/shorten?url=${encodeURIComponent(url)}`);
+
+    if (response.ok) {
+      const data = (await response.json()) as { shortUrl: string };
+      return data.shortUrl;
+    }
+  } catch (e) {}
```
Comment (nickytonline) [inline lib/utils/shorten-url.ts:9]: consider logging an explicit error to Sentry if URL shortening doesn't work. Also, if URL shortening fails, should we fallback to the full URL?

Comment (brandonroberts) [reply lib/utils/shorten-url.ts:9]: Good idea about Sentry.

And yes, if the request fails or the API is down, it falls back to the original URL

**Comment by bdougie** [comment]
This works beautifully. Curious if there will be a follow up PR to make oss.fyi work. I see the link is dub.sh currently.

**Comment by brandonroberts** [comment]
@bdougie that will be handled in the API, didn't want to create a bunch of oss.fyi links in development mode

**Review by bdougie — APPROVED** [review]
Love it

**Review by isabensusan — APPROVED** [review]
Tested out with multiple profles and it works perfectly. LGTM!

**Review by jpmcb — APPROVED** [review]

**Pull request merged**